### DISCUSSION TITLE
BlenderGIS 2.2.8 and Blender 4.0

### DISCUSSION THREAD

**Discussion by Turbito61** [domlysz/BlenderGIS#833] [General]
When trying out Blender 4.0, BlenderGIS 2.2.8 was imported from v3.6.5 and was accessible from the menu. But when importing a raster file as Basemap on a new plane", it threw a ValueError("1-2 args execution context is supported") exception in Blender python code ops.py, line 60, when being called from BlenderGIS, bgis_utils.py, line 107: bpy.ops.view3d.view_selected(overrideContext).

Comparing Blender source code 3.6.5 against 4.0.0 you can see, that Blender 4.0 now refuses to take the dict argument overrideContext, in fact they removed the code, which handled it due to being deprecate.

As a quick-and-dirty fix for me, I simply removed the overrideContext argument, in spite of likely losing functionality here, but it helped me so far to run the BlenderGIS addon in Blender 4.0. I ran a quick test and it behaves as expected for the things I do with it. Perhaps there are still errors in other functions. This is what I could do since I was unable to fix it in a clean way. I never before digged into any Blender (addon) source code. Maybe someone more experienced in Blender stuff can do it better.

EDIT: Of course I was using the 3.6.5 LTS version before. Corrected.

**Comment by arijit08** [comment]
Could you please tell me how to remove the argument at our end as well?

**Reply by Turbito61** [reply]
The corresponding code in BlenderGIS 2.2.8:

operators/utils/bgis_utils.py, line 103-107:

			if zoomToSelect:
				overrideContext = context.copy()
				overrideContext['area'] = area
				overrideContext['region'] = area.regions[-1]
				bpy.ops.view3d.view_selected(overrideContext)

I simply removed the parameter "overrideContext" from the call in the last line. I could do it, because it is optional. Of course I removed the functionality behind this as well. As far as I understood it is something like focusing and zooming into the new created plane in Blender. I decided that I (myself) could do without it for now for the sake of running it in Blender 4.0.

As you can see, the "overrideContext" parameter is a dict parameter and starting from Blender v4.0, dict parameters are not accepted anymore to the bpy.ops.view3d.view_selected() call. They were completely removed from the interface here.

I'm not familiar with the Blender operator concept, therefore I can't see what Blender 3.6 did with the parameter. ATM I simply don't find the code that was handling it.

Someone who understands what was happening in Blender 3.6 at this point should emulate the same functionality in BlenderGIS according to the interface of Blender 4.0. Unfortunately ATM I can't help here any further.

**Reply by arijit08** [reply]
Thank you so much!

**Reply by Niuskir** [reply]
Worked for me. Thks!

**Reply by Winter-And-You-Gone** [reply]
Thanks！Indeed effective, I hope the GIS team can resolve this compatibility issue as soon as possible

**Reply by louqualen** [reply]
How do I edit the underlying code of the add-on itself? I cannot find "operators/utils/bgis_utils.py" in the text editor after enabling dev tools

**Reply by Turbito61** [reply]
Windows:
If you have installed BlenderGIS 2.2.8, you'll find its source code under
C:\Users\UserName\AppData\Roaming\Blender Foundation\Blender\4.0\scripts\addons\BlenderGIS-228

**Comment by phonpitawas** [comment]
it's work thanks bro.

**Comment by domlysz** [comment]
fixed https://github.com/domlysz/BlenderGIS/releases/tag/229
